metrics/firebase/docs/features/builds_aggregation/01_firestore_builds_aggregation_design.md

```diff
@@ -0,0 +1,176 @@
+
+```dart
+functions['onBuildAdded'] = functions.firestore
+    .document('build/{buildId}')
+    .onCreate(onBuildAddedHandler)
+
+
+Future<void> onBuildAddedHandler(DocumentSnapshot snapshot, EventContext context) {...}
+```
+
+The trigger's `onBuildAddedHandler` handler should process incrementing logic for the `build_days` collection document, based on the created build's status and started date.
+
+The following sequence diagram shows the overall process of how the `onCreate` triggers works:
+
+![Sequence diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/platform-platform/monorepo/builds_aggregation_design/metrics/firebase/docs/features/builds_aggregation/diagrams/firestore_create_builds_aggregation_sequence_diagram.puml)
+
+#### onUpdate trigger
+
+> Explain the main purpose of the trigger.
+
+The second trigger - `onUpdate`, should handle the logic to increment or decrement the builds count, related to changes in the build status. For example, if the build with an `inProgress` status changes to `successful`, we should increment `successful` count of the document, and decrement the `inProgress` count.
+
+```dart
+functions['onBuildUpdated'] = functions.firestore
+    .document('build/{buildId}')
+    .onUpdate(onBuildUpdatedHandler)
+
+Future<void> onBuildUpdatedHandler(Change<DocumentSnapshot> change, EventContext context) {...}
+```
+
+In case, the `onCreate` or `onUpdate` trigger's handler processing failed, we should create a new document inside the `tasks` collection. Later, we can use this collection to fix the `build_days` counters.
+
+The following sequence diagram shows the overall process of how the `onUpdate` triggers works:
+
+![Sequence diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/platform-platform/monorepo/builds_aggregation_design/metrics/firebase/docs/features/builds_aggregation/diagrams/firestore_update_builds_aggregation_sequence_diagram.puml)
+
+#### Testing
+
+> How will the functions be tested?
+
+The described above functions we will test using the [test](https://pub.dev/packages/test) package, and perform unit-testing the functions' event handlers.
+
+The `onCreate` handler takes two arguments - `DocumentSnapshot` and `EventContext`. As we don't want to reference the real instance of Firestore, we need to "mock" the `DocumentSnapshot` using the [mockito](https://pub.dev/packages/mockito) package. This gives us a possibility to emulate a Firestore and return specific results depending on the test. Also, we should mock the `EventContext` and pass that instance as the second parameter to the handler.
+
+The `OnUpdate` event handler, has a similar second argument, but the first one is a `Change`. As it has two states: after the update event and prior to the event, so we should create an instance of `Change` class using the "mocked" `DocumentSnapshot`s and pass it as the first argument to the handler.
```

Comment:
Would it be great to add information on how are we going to mock the `FirebaseAdmin` instance to test that out cloud function actually writes or updates documents?